scripts/swupdate/merge.py

```diff
@@ -0,0 +1,104 @@
+        'configFile' : "/opt/seagate/cortx/auth/resources/authserver.properties",
+        'oldSampleFile' : "/tmp/authserver.properties.sample.old",
+        'newSampleFile' : "/opt/seagate/cortx/auth/resources/authserver.properties.sample",
+        'unsafeAttributesFile' : "/opt/seagate/cortx/auth/resources/authserver_unsafe_attributes.properties",
+        'fileType' : 'properties://'
+    },
+    'keystore' : {
+        'configFile' : "/opt/seagate/cortx/auth/resources/keystore.properties",
+        'oldSampleFile' : "/tmp/keystore.properties.sample.old",
+        'newSampleFile' : "/opt/seagate/cortx/auth/resources/keystore.properties.sample",
+        'unsafeAttributesFile' : "/opt/seagate/cortx/auth/resources/keystore_unsafe_attributes.properties",
+        'fileType' : 'properties://'
+    },
+    'bgdelete' : {
+        'configFile' : "/opt/seagate/cortx/s3/s3backgrounddelete/config.yaml",
+        'oldSampleFile' : "/tmp/config.yaml.sample.old",
+        'newSampleFile' : "/opt/seagate/cortx/s3/s3backgrounddelete/config.yaml.sample",
+        'unsafeAttributesFile' : "/opt/seagate/cortx/s3/s3backgrounddelete/s3backgrounddelete_unsafe_attributes.yaml",
+        'fileType' : 'yaml://'
+    }
+}
+
+def upgrade_config(configFile:str, oldSampleFile:str, newSampleFile:str, unsafeAttributesFile:str, filetype:str):
+    """
+    Core logic for updating config files during upgrade using conf store.
+    """
+    #If config file is not present then simply copy sample file.
+    if not os.path.isfile(configFile):
+        shutil.copy(newSampleFile, configFile)
+        return
+
+    conf_file =  filetype + configFile
+    conf_old_sample = filetype + oldSampleFile
+    conf_new_sample = filetype + newSampleFile
+    conf_unsafe_file = filetype + unsafeAttributesFile
+
+    cs_conf_file = S3CortxConfStore(config=conf_file, index=conf_file)
+    cs_conf_old_sample = S3CortxConfStore(config=conf_old_sample, index=conf_old_sample)
+    cs_conf_new_sample = S3CortxConfStore(config=conf_new_sample, index=conf_new_sample)
+    cs_conf_unsafe_file = S3CortxConfStore(config=conf_unsafe_file, index=conf_unsafe_file)
+
+    conf_file_keys = cs_conf_file.get_all_keys()
+    conf_unsafe_file_keys = cs_conf_unsafe_file.get_all_keys()
+    conf_new_sample_keys = cs_conf_new_sample.get_all_keys()
+
+    #logic to determine which keys to merge.
+    keys_to_overwrite = []
+    for key in conf_new_sample_keys:
+        #check if the key is safe for modification.
+        if key in conf_unsafe_file_keys:
+            continue
+        #if key not present in config then add it to config.
+        if key not in conf_file_keys:
+            keys_to_overwrite.append(key)
+        #if config value == old sample then change to new value.
+        elif cs_conf_file.get_config(key) == cs_conf_old_sample.get_config(key):
+            keys_to_overwrite.append(key)
+        #if user has changed the value of this key then skip it.
+        else:
+            continue
+
```

Comment:
Add this logic before for loop in a special function. Write comments 
1)search for keys with [] in config 
2)delete these keys/values in config